### PR TITLE
Update AWS Terraform deployment example

### LINKED ISSUE #137: AWS ECS Fargate Terraform - Feedback and tweaks
Today I tried the ECS + Fargate Terraform

## Feedback

First of all, excellent work. It was a breeze to setup. It worked the first time, even after several changes I did to match our internal best practices, it worked like a charm.

## Suggested tweaks

- Tags: Most organizations have a well defined taxonomy for tags. Define a `var.tags` variable that users can populate, and set it in all the resources supporting it.
- Default name prefix: Define variable `var.name_prefix`, and use it in all the resources where some name, name prefix or id is required:
- Specify the retention days of the log group

``` text
resource "aws_cloudwatch_log_group" "scim-bridge" {
  name = local.name_prefix
  tags = local.tags
  retention_in_days = 14
}

resource "aws_ecs_cluster" "scim-bridge" {
  name = local.name_prefix
  tags = local.tags
}

...
```

- Obtain the Zone ID from a data source: Instead of taking the Route53 Zone ID as a variable, use the domain to determine the Zone ID:

``` text
data "aws_route53_zone" "zone" {
  name         = var.domain // var.domain = example.com
  private_zone = false
}
```

- Obtain the VPC and subnets from a data source: Most organizations don't use the default VPC, and there's a VPC already setup.

``` text
# Find the vpc by name
data "aws_vpc" "vpc" {
  tags = {
    Name = "my-vpc-name"
  }
}

# Find the public subnets in the VPC
data "aws_subnet_ids" "subnets" {
  vpc_id = data.aws_vpc.vpc.id
  tags = { SubnetTier = "public"}
}
```

- Specify the VPC id when setting up the sec. groups:

``` text
resource "aws_security_group" "scim-bridge-sg" {
  vpc_id = data.aws_vpc.vpc.id
...
}
```

- Get the ACM certificate from a data source instead of creating a new one: Most organizations already have an ACM cert "*.example.com"

``` text
data "aws_acm_certificate" "scim_certificate" {
  domain = "*.${var.domain}"
}
```

### PR DESCRIPTION
**Pull request by ag-adampike**
This PR incorporates several suggestions from #137:
* Tags can be added to all supported AWS resources.
* A common prefix can be applied to the name of all supported AWS resources.
* A specific log retention period can be specified.
* The Route 53 Zone ID is populated automatically using the domain of the DNS record.
* An VPC can be specified instead of using the default VPC for the region.
* The deployment can optionally make use of an existing wildcard certificate rather than creating a unique record for the SCIM bridge.

I also cleaned up some of the code to adhere to some Terraform best practices, mainly using the [Terraform Best Practices](https://www.terraform-best-practices.com/) site as a reference:
* cleaned up some indents and spacing throughout
* reordered the internal structure of the blocks for consistency and clarity
* renamed the main `terraform.tf` file to `main.tf`
* moved the `output` blocks to a separate file (`outputs.tf`)
* added variable descriptions

Finally, I updated the `README` to remove any conflicting instructions and explain how to use the new functionality.

P.S. it's not noted on any commit, but I would like to recognize @nathan1p and @notjuuke for their contributions to this work. :-)

Fixes #137.

### CODE REVIEW

File: aws-ecsfargate-terraform/README.md
Comment (devillexio) [inline aws-ecsfargate-terraform/README.md:44]: I'm wondering if this is more clear than the previous version. Can I ask why we changed it?

File: aws-ecsfargate-terraform/README.md
Comment (devillexio) [inline aws-ecsfargate-terraform/README.md:57]: In a similar vein, I understand the the change to `scim.example.com` but I'm not sure we should wrap it in `<>`.

File: aws-ecsfargate-terraform/README.md
Comment (devillexio) [inline aws-ecsfargate-terraform/README.md:20]: I think it may be worth mentioning any default config values when applicable in this section.

**Review by devillexio — COMMENTED** [review]
Looking great so far! 🙌 Minor feedback all around.

File: aws-ecsfargate-terraform/variables.tf
Comment (ag-adampike) [reply aws-ecsfargate-terraform/variables.tf:3]: Thanks! Going by [our style guide](https://support.1password.com/style-guide/#1password-scim-bridge) on this one, it's okay to shorten to SCIM bridge after the first occurence, but I guess I did it out of order. 🙃

File: aws-ecsfargate-terraform/terraform.tfvars.template
Comment (ag-adampike) [reply aws-ecsfargate-terraform/terraform.tfvars.template:16]: In my mind, I'm differentiating between using a `#` for literally making a comment (where readability is important) and using it ignore a line (where I just want to delete the character to "switch" the line on).

l'm happy defer to you on this. :-)

File: aws-ecsfargate-terraform/variables.tf
Comment (ag-adampike) [reply aws-ecsfargate-terraform/variables.tf:40]: This variable is used to set the `retention_in_days` in the `aws_cloudwatch_log_group` resource block. If not specified for this resource, or if the value is set to 0, logs are kept indefinitely.

I've set it as 0 in the the `terraform.tfvars.template` file to reduce the amount of work needed to override the default, and make it intuitive. To override the default behaviour, change this value to something other than `0` (instead of having to add the variable manually or uncomment a line and add a value.

File: aws-ecsfargate-terraform/variables.tf
Comment (ag-adampike) [reply aws-ecsfargate-terraform/variables.tf:18]: My bad. I fixed this in the README by using "common" instead of "default", but I mean some string of text to prepend to the name of each resource. Essentially, the default is nothing, and they just get named how we name them in the Terraform script. If they want to, they can use this variable to tack something else onto the front of those names.

File: aws-ecsfargate-terraform/README.md
Comment (ag-adampike) [reply aws-ecsfargate-terraform/README.md:9]: It's in the root of the repo: https://github.com/1Password/scim-examples/blob/master/PREPARATION.md. The hyperlink seems to work correctly for me. 🤔

File: aws-ecsfargate-terraform/README.md
Comment (ag-adampike) [reply aws-ecsfargate-terraform/README.md:12]: LGTM

File: aws-ecsfargate-terraform/README.md
Comment (ag-adampike) [reply aws-ecsfargate-terraform/README.md:44]: I'm trying to signify a placeholder, highlighting exactly what characters need to be replaced, so people don't blindly copy/paste, or replace any more or less than they need to. Angle brackets are pretty intuitive to me for this, but I will happily defer to our best practices for this. :-)

File: aws-ecsfargate-terraform/README.md
Comment (ag-adampike) [reply aws-ecsfargate-terraform/README.md:57]: (See [above](https://github.com/1Password/scim-examples/pull/169#discussion_r788213623). 👆🏻)

File: aws-ecsfargate-terraform/README.md
Comment (ag-adampike) [reply aws-ecsfargate-terraform/README.md:20]: Happy to add something here. Can you elaborate?

File: aws-ecsfargate-terraform/terraform.tfvars.template
Comment (devillexio) [reply aws-ecsfargate-terraform/terraform.tfvars.template:16]: No problem. We can leave it as is. I realized that I'm biased based on how VS Code comments out lines when you hit `CMD + /`.

We can leave it as is.

File: aws-ecsfargate-terraform/variables.tf
Comment (devillexio) [reply aws-ecsfargate-terraform/variables.tf:3]: Ok cool. Thanks for confirming. I'm happy either way. Feel free to resolve this.

File: aws-ecsfargate-terraform/variables.tf
Comment (devillexio) [reply aws-ecsfargate-terraform/variables.tf:40]: Ah thanks for the clarification. Makes sense to me. :+1:

File: aws-ecsfargate-terraform/README.md
Comment (devillexio) [reply aws-ecsfargate-terraform/README.md:9]: Ah, it's I know why the link was broken. It's because I viewed the rich edit view of the diff. Thanks for checking.

File: aws-ecsfargate-terraform/README.md
Comment (devillexio) [reply aws-ecsfargate-terraform/README.md:44]: Cool! Happy with it as is. As long as it doesn't work when you copy and paste and forget to update it, I'm happy. 🙂

File: aws-ecsfargate-terraform/README.md
Comment (devillexio) [reply aws-ecsfargate-terraform/README.md:20]: I thought it may be worthwhile saying if any config values under this section have a defined default value or not, and what it is. That being said reading through the section again I realize many of the values are optional and already have a detailed description.

I'm happy with it as is.

**Review by devillexio — DISMISSED** [review]
Looking good! 🙌 

I'd still like to give @alicethorne-ab an opportunity to also give the changes a review.

File: .gitignore
Comment (alicethorne-ab) [inline .gitignore:12]: Very minor nitpick: I'd shorten this to `*.plan`.

**Review by alicethorne-ab — DISMISSED** [review]
Looks good to me! :shipit:

File: .gitignore
Comment (ag-adampike) [reply .gitignore]: Done.

**Comment by ag-adampike** [comment]
Made a few final tweaks based on your review and got those commits signed with my GPG key, so this should be good to merge. Thanks for the review!

**Review by ag-adampike — COMMENTED** [review]
Made a few final tweaks based on your reviews and got those commits signed with my GPG key, so this should be good to merge. Thanks for the review!

**Review by devillexio — APPROVED** [review]
Re-approving after commits were signed.

**Pull request merged**
